### PR TITLE
feat(client): auto reuse files of model

### PR DESCRIPTION
**Pull request by jialeicui**
## Description

Auto reuse file of the model (or dataset)
- Remove the model field from the model.yaml
- Decide whether to package into tar based on file size (currently set at 10K threshold).
- Automatically use files from object store when building or copying from remote.
- Default to using hard links (friendly for tar or various operations).

[asciicast](https://asciinema.org/a/FG6V3FKqiXSBoNHEGwMwpkwMN)

## Modules
- [ ] UI
- [ ] Controller
- [ ] Agent
- [x] Client
- [ ] Python-SDK
- [ ] Others

## Checklist
- [x] run code format and lint check
- [x] add unit test
- [ ] add necessary doc

### CODE REVIEW

File: client/starwhale/core/model/model.py [hunk 1]
```diff
@@ -651,23 +648,29 @@ def buildImpl(self, workdir: Path, **kw: t.Any) -> None:  # type: ignore[overrid
...
-                    }
-                )
+                size = file_stat(self.store.src_dir / sub_path).st_size
+                separate = size > size_th_to_tar
+                file_info = {
+                    "name": os.path.basename(f),
+                    "path": f"{self.store.src_dir_name}/{sub_path}",
+                    "signature": blake2b_file(self.store.src_dir / sub_path),
+                    "duplicate_check": separate,
+                    "desc": FileDesc.SRC.name if not separate else FileDesc.MODEL.name,
```
Comment (goldenxinxing) [inline client/starwhale/core/model/model.py:666]: How about redefine the FIleDesc? such as : LargeFile, SmallFile

**Comment by jialeicui** [comment]
Closes #1798

**Comment by tianweidut** [comment]
update docs and `model.yaml` in the examples?

others LGTM

**Comment by jialeicui** [comment]
> update docs and `model.yaml` in the examples?
> 
> others LGTM

Done https://github.com/star-whale/starwhale/pull/1906/commits/87a1e76dda5509111838574ae00d98e051c53647

@tianweidut

**Review by tianweidut — APPROVED** [review]

**Pull request merged**